Integration and refactorings

ManfredKarrer opened the pull request:
I did a bunch of refactorings. 
I added gradle but for the tests there are some issue with dependencies. I had not run any tests but run both headless and Desktop and it seems to work.
You need to update your core and desktop projects with my branches:
https://github.com/ManfredKarrer/bisq-core/tree/http-api
https://github.com/ManfredKarrer/bisq-desktop/tree/http-api

I used that repo (by accident) as well:
https://github.com/ManfredKarrer/bisq-http-api/tree/http-api
It is identical with https://github.com/ManfredKarrer/bisq-api but carries the correct name.

Commit:
WIP: core integration

ManfredKarrer committed:
Manfreds changes

ManfredKarrer committed:
Add gradle files

ManfredKarrer committed:
Setup refactorings

- Remove ApiEnvironment
- ApiMain: Remove setupEnvironment call
- ApiModule: Add httpApiHost and httpApiPort, remove BtcAddressValidator
and ApiEnvironment
- Api: Rename shutdown to hostShutdownHandler
- BisqApiApplication: Add constructor and use Injector there, use
private final fields.
Rename shutdown to hostShutdownHandler.
Add shutdownHandler for handling local shutdown procedures.
Add startServer method.
Rename setupCors to setupCrossOriginFilter.
- BisqProxy: Rename shutdown to shutdownHandler

ManfredKarrer committed:
HttpApi refactorings

- Remove ApiOptionCustomizer and ApiOptionKeys
- Rename ApiMain to HttpApiMain
- Rename StandaloneApiModule to HttpApiHeadlessModule
- Rename Api to HttpApiHeadlessApp
- Rename BisqApiApplication to HttpApiServer
- Rename ApiModule to HttpApiModule

- Use BisqHeadlessApp and BisqHeadlessAppMain base classes to avoid
duplicated code.

- Move initialisation of HttpApiServer in BisqAppMain from
onApplicationStarted to onSetupComplete. That way the UI has more time
to render and we prevent a gray screen while HttpApiServer is starting
up.

ManfredKarrer committed:
HttpApi refactorings

- Move auth classes to auth package
- Move auth ResourceHelper to util package

ManfredKarrer committed:
Rename package: network.bisq.api to network.bisq.httpapi

ManfredKarrer committed:
Remove sub package network from network.bisq

- Rename test packages

ManfredKarrer committed:
Use Guice constructor injection instead of injector.getInstance

ManfredKarrer committed:
Move excetpion classes to excep. package

ManfredKarrer committed:
Refactorings

- Remove BtcAddressValidator
- Move CurrencyListHealthCheck to other package

ManfredKarrer committed:
Move ApiV1 one level up

ManfredKarrer committed:
Use Guice for ApiV1

ManfredKarrer committed:
Refactorings

- Rename ApiV1 to HttpApiInterfaceV1
- Use Guice for resource classes and HttpApiInterfaceV1
- Move CurrencyListHealthCheck to util package

ManfredKarrer committed:
Rename bisq.httpapi.service.v1 to bisq.httpapi.service.resources

ManfredKarrer committed:
Add comment

ManfredKarrer committed:
Refactorings

- Use BalanceModel to get balances
- Rename WalletDetails to Balances

ManfredKarrer committed:
Remove BisqProxy.getWalletDetails

ManfredKarrer committed:
Refactor BisqSetup

- Rename BalanceModel to Balances
- Move listener setup code for balances to Balances.onAllServicesInitialized
- Move tradeManager init code to tradeManager.onAllServicesInitialized

- Use Balances for accessing balance
- Add BalanceUtil in views and api to avoid code duplication
- Use Balances in WalletResource
- Use BalanceUtil in BisqProxy

ManfredKarrer committed:
Refactor BisqProxy

- Use OfferBookService in OfferResource to access offers

ManfredKarrer committed:
Move offer specific code to OfferResource

ManfredKarrer committed:
Refactor OfferResource

- Add check for isBootstrapped
- Make OfferList.getTotal a method
- Add handleException and completeExceptionally to ResourceHelper

ManfredKarrer committed:
Refactor OfferResource, Fix incorrect currency pair

- Rename OfferToCreate to InputDataForOffer
- Move getMarketList and getCurrencyList to MarketResource
- Rename checkMarketValidity to validateMarketPair

blabno commented:
I've checked out your bisq-api and core branches and run
 `./gradlew --include-build ../bisq/core test`
But I'm getting:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/bernard/projects/bisq/bisq/core/build.gradle' line: 59

* What went wrong:
A problem occurred evaluating project ':bisq-core'.
> Could not find method testAnnotationProcessor() for arguments [org.projectlombok:lombok:1.16.16] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

```

blabno commented:
Maybe let's first refactor the code using maven, so that we're sure that tests are passing and then migrate to gradle?

ManfredKarrer committed:
Refactor OfferBuilder. Remove code duplication

- Remove duplicated and outdated code for creating an offer and use
methods from OfferUtil instead which encapsulates the functionalities
for GUI and API.
- Add TxFeeEstimation class (TODO at GUI we have nor replaced it yet
as it requires more testing to be sure that all works as before).

ManfredKarrer commented:
In my setup with the other projects I cannot mix it with maven. But feel free to do it on your side as it works best for you. At some point we need to get gradle working when we want to integrate it....

Commit:
Migrate from maven to gradle

mrosseel committed:
Merge pull request #108 from blabno/feature/gradle

Feature/gradle

Commit:
Fix tests

Commit:
Add F2F payment account

Commit:
Static analysis tweaks

Commit:
HalCash payment account

Commit:
MoneyGram payment account

Commit:
WeChatPay payment account

ManfredKarrer committed:
Merge branch 'http-api-with-gradle' into http-api

# Conflicts:
#	.travis.yml
#	Readme.md
#	build.gradle
#	docker/startApi.sh
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	pom.xml
#	src/test/java/bisq/httpapi/ApiTestHelper.java
#	src/test/java/bisq/httpapi/ArbitratorResourceIT.java
#	src/test/java/bisq/httpapi/BackupResourceIT.java
#	src/test/java/bisq/httpapi/BitcoinIT.java
#	src/test/java/bisq/httpapi/ClosableTradableResourceIT.java
#	src/test/java/bisq/httpapi/ContainerFactory.java
#	src/test/java/bisq/httpapi/CurrencyResourceIT.java
#	src/test/java/bisq/httpapi/OfferResourceIT.java
#	src/test/java/bisq/httpapi/PaymentAccountIT.java
#	src/test/java/bisq/httpapi/PreferencesResourceIT.java
#	src/test/java/bisq/httpapi/RegexMatcher.java
#	src/test/java/bisq/httpapi/TradeResourceIT.java
#	src/test/java/bisq/httpapi/UserResourceIT.java
#	src/test/java/bisq/httpapi/VersionResourceIT.java
#	src/test/java/bisq/httpapi/WalletResourceIT.java
#	src/test/java/bisq/httpapi/arquillian/CubeLogger.java
#	src/test/java/bisq/httpapi/arquillian/CubeLoggerExtension.java
#	src/test/java/network/bisq/api/ApiTestHelper.java
#	src/test/java/network/bisq/api/ArbitratorResourceIT.java
#	src/test/java/network/bisq/api/BackupResourceIT.java
#	src/test/java/network/bisq/api/BitcoinIT.java
#	src/test/java/network/bisq/api/ClosableTradableResourceIT.java
#	src/test/java/network/bisq/api/ContainerFactory.java
#	src/test/java/network/bisq/api/CurrencyResourceIT.java
#	src/test/java/network/bisq/api/OfferResourceIT.java
#	src/test/java/network/bisq/api/PaymentAccountIT.java
#	src/test/java/network/bisq/api/PreferencesResourceIT.java
#	src/test/java/network/bisq/api/RegexMatcher.java
#	src/test/java/network/bisq/api/TradeResourceIT.java
#	src/test/java/network/bisq/api/UserResourceIT.java
#	src/test/java/network/bisq/api/VersionResourceIT.java
#	src/test/java/network/bisq/api/WalletResourceIT.java
#	src/test/java/network/bisq/api/arquillian/CubeLogger.java
#	src/test/java/network/bisq/api/arquillian/CubeLoggerExtension.java
#	src/test/resources/META-INF/services/org.jboss.arquillian.core.spi.LoadableExtension
#	src/testIntegration/java/bisq/httpapi/ApiTestHelper.java
#	src/testIntegration/java/bisq/httpapi/ArbitratorResourceIT.java
#	src/testIntegration/java/bisq/httpapi/BackupResourceIT.java
#	src/testIntegration/java/bisq/httpapi/BitcoinIT.java
#	src/testIntegration/java/bisq/httpapi/ClosableTradableResourceIT.java
#	src/testIntegration/java/bisq/httpapi/ContainerFactory.java
#	src/testIntegration/java/bisq/httpapi/CurrencyResourceIT.java
#	src/testIntegration/java/bisq/httpapi/OfferResourceIT.java
#	src/testIntegration/java/bisq/httpapi/PaymentAccountIT.java
#	src/testIntegration/java/bisq/httpapi/PreferencesResourceIT.java
#	src/testIntegration/java/bisq/httpapi/RegexMatcher.java
#	src/testIntegration/java/bisq/httpapi/TradeResourceIT.java
#	src/testIntegration/java/bisq/httpapi/UserResourceIT.java
#	src/testIntegration/java/bisq/httpapi/VersionResourceIT.java
#	src/testIntegration/java/bisq/httpapi/WalletResourceIT.java
#	src/testIntegration/java/bisq/httpapi/arquillian/CubeLogger.java
#	src/testIntegration/java/bisq/httpapi/arquillian/CubeLoggerExtension.java

ManfredKarrer committed:
Fix wrong currencyCode handling

ManfredKarrer committed:
Use UserThread.execute

Rename OfferResource to OfferEndPoint

Commit:
Change Resource suffix to Endpoint

Commit:
Rename resources package to endpoint

Commit:
Revert Manfred changes of build.gradle.

Commit:
Change expected maxTradeLimit and restore check of accountAgeWitnessHash.

Commit:
Change Resource suffix to Endpoint

Commit:
Introduce OfferFacade

Commit:
Fix resourcePackage in bisq-api.yml

Commit:
Surround all calls in OfferEndpoint with UserThread

Commit:
Introduce TradeFacade

Commit:
Surround all calls in TradeEndpoint with UserThread

Commit:
Introduce PaymentMethodFacade

Commit:
Surround all calls in PaymentAccountEndpoint with UserThread

Commit:
Introduce WalletFacade and ShutdownFacade

Commit:
Surround all calls in WalletEndpoint with UserThread

Commit:
Introduce BackupFacade

Commit:
Surround all calls in BackupEndpoint with UserThread

Commit:
Introduce ArbitratorFacade

Commit:
Surround all calls in ArbitratorEndpoint with UserThread

Commit:
Introduce PreferencesEndpoint

Commit:
Surround all calls in PreferencesEndpoint with UserThread

Commit:
Introduce ClosedTradableEndpoint

Commit:
Surround all calls in ClosedTradableEndpoint with UserThread

Commit:
Introduce UserEndpoint

Commit:
Surround all calls in UserEndpoint with UserThread

Commit:
Introduce NetworkEndpoint

Commit:
Surround all calls in NetworkEndpoint with UserThread

Commit:
Remove version method from BisqProxy

Commit:
Introduce PriceFeedFacade

Commit:
Surround all calls in CurrencyEndpoint with UserThread
